quantizer.cpp: fix cuda memory pinning

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#41139 quantizer.cpp: fix cuda memory pinning**

Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22438787](https://our.internmc.facebook.com/intern/diff/D22438787)

vkuzo committed:
quantizer.cpp: fix cuda memory pinning

Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

ngimel reviewed (approved):
Approving, if cpu-only builds pass

vkuzo committed:
Update on "quantizer.cpp: fix cuda memory pinning"


Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "quantizer.cpp: fix cuda memory pinning"


Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22438787](https://our.internmc.facebook.com/intern/diff/D22438787)

[ghstack-poisoned]

ngimel commented:
Can you please also add a test case to the test suite to make sure memory is not pinned?

vkuzo committed:
add test on "quantizer.cpp: fix cuda memory pinning"


Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22438787](https://our.internmc.facebook.com/intern/diff/D22438787)

[ghstack-poisoned]

ngimel commented:
test_quantization time is down to 665 s from more than a 1000. 🎉

vkuzo committed:
Update on "quantizer.cpp: fix cuda memory pinning"


Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22438787](https://our.internmc.facebook.com/intern/diff/D22438787)

[ghstack-poisoned]

vkuzo committed:
Update on "quantizer.cpp: fix cuda memory pinning"


Summary:

Fixes the test case in https://github.com/pytorch/pytorch/issues/41115
by using PyTorch's CUDA allocator instead of the old Caffe2 one.

Test Plan:

run the test case from the issue:
https://gist.github.com/vkuzo/6d013aa1645cb986d0d4464a931c779b

let's run CI and see what it uncovers

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22438787](https://our.internmc.facebook.com/intern/diff/D22438787)

[ghstack-poisoned]
